src/Bicep.Core/Workspaces/SourceFileGrouping.cs

```diff
@@ -7,10 +7,31 @@
 using Bicep.Core.Extensions;
 using Bicep.Core.FileSystem;
 using Bicep.Core.Navigation;
+using Bicep.Core.Syntax;
 using static Bicep.Core.Diagnostics.DiagnosticBuilder;
 
 namespace Bicep.Core.Workspaces
 {
+    public interface IArtifactResolutionInfo
+    {
+        IForeignTemplateReference ForeignTemplateReference { get; }
+        ISourceFile ParentTemplateFile { get; }
+    }
+    public record ModuleSourceResolutionInfo(
+        IForeignTemplateReference ModuleDeclaration,
+        ISourceFile ModuleParentTemplateFile) : IArtifactResolutionInfo
+    {
+        public IForeignTemplateReference ForeignTemplateReference => ModuleDeclaration;
+        public ISourceFile ParentTemplateFile => ModuleParentTemplateFile;
+    }
+
+    public record ProviderSourceResolutionInfo(
```

Comment:
>ProviderSourceResolutionInfo [](http://example.com/codeflow?start=18&length=28)

Other than parameter/property names, `ProviderSourceResolutionInfo` seems to be identical to `ModuleSourceResolutionInfo`

Reply:
Yes, I am using the type information later in the processing to handle providers in a different way than modules